lib/preprocessor.js

```diff
@@ -146,17 +145,19 @@ function createCoveragePreprocessor (logger, basePath, reporters = [], coverageR
         log.error('%s\n  at %s', err.message, file.originalPath)
         done(err.message)
       } else {
-        if (file.sourceMap && instrumenter.lastSourceMap()) {
+        // Use the incoming source map, or the one generated from the instrumenter (if configured to generate one)
+        const sourceMap = file.sourceMap || instrumenter.lastSourceMap()
+
+        if (sourceMap) {
           log.debug('Adding source map to instrumented file for "%s".', file.originalPath)
-          const generator = SourceMapGenerator.fromSourceMap(new SourceMapConsumer(instrumenter.lastSourceMap().toString()))
-          generator.applySourceMap(new SourceMapConsumer(file.sourceMap))
-          file.sourceMap = JSON.parse(generator.toString())
```

Comment:
it will potential broke karma reporter.js has check on file.sourceMap but file.sourceMap - never set

Reply:
Could you provide a link? As far as I'm aware, there are no checks in this plugin's reporter.js, but I'm not as familiar with the greater `karma` codebase.

If it's needed, I could set `file.sourceMap` to the local variable `sourceMap` (defined [here](https://github.com/karma-runner/karma-coverage/pull/394/files#diff-1006b0ebc64f4a0530e73029eed686b0R149).

Reply:
Karma [reporter](https://github.com/karma-runner/karma/blob/adc6a66943c93185e8481060c3cf4b7416985151/lib/reporter.js#L55) 

And the same line I found in [karma-typescript](https://github.com/monounity/karma-typescript/blob/f9dee897c69a1cf49586561e9c0637f854c9083b/packages/karma-typescript/src/bundler/bundler.ts#L91)

Reply:
Thanks for the link -- I wasn't aware of `file.sourceMap` being used for remapping stacktraces when errors get thrown!

If that's the case, I might need to add this source map merging code back in since karma will use the instrumented js as the source that needs to get mapped all the way back to the original code (potentially past transpiled JS back to TS or coffeescript).

Reply:
I believe this should be resolved from my latest commit.